### DISCUSSION TITLE
What is the reason for not handling throwables in `resolveType` while they are handled in `resolve`?

### DISCUSSION THREAD

**Discussion by ruudk** [webonyx/graphql-php#1144] [Q&A]
I noticed something interesting.

When one of my `resolve` callbacks throws an exception, they are handled properly, and the Promise is rejected.

But when an exception handles in `resolveType` the error is not handled and just bubbles up.

Is there a particular reason for that? 

Wouldn't it be nice if a failure in `resolveType` would just reject the promise?

**Comment by ruudk** [comment]
@spawnia do you maybe have an idea? Should we fix it?

**Reply by spawnia** [reply]
> the error is not handled and just bubbles up

What do you mean by that? Is it not caught by the usual error handling facilities of this library and just causes a partial failure of a subtree of the query?
